app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -413,6 +390,46 @@ class PasswordStore : AppCompatActivity() {
         createRepository()
     }
 
+    /**
+     * Validates if storage permission is granted, and requests for it if not. The return value
+     * indicates whether or not a permission request is being created.
+     */
+    private fun checkStoragePermissions(checkLocalRepo: Boolean = false): Boolean {
+        return if (ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                val snack = Snackbar.make(
+                        findViewById(R.id.main_layout),
+                        getString(R.string.access_sdcard_text),
+                        Snackbar.LENGTH_INDEFINITE)
+                        .setAction(R.string.dialog_ok) {
+                            ActivityCompat.requestPermissions(
+                                    activity,
+                                    arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+                                    REQUEST_EXTERNAL_STORAGE)
+                        }
+                snack.show()
+                val view = snack.view
+                val tv: AppCompatTextView = view.findViewById(com.google.android.material.R.id.snackbar_text)
+                tv.setTextColor(Color.WHITE)
```

Comment:
Is it normal to hack on Snackbars in this way?

Reply:
> Is it normal to hack on Snackbars in this way?

Nope. Setting the action is usually enough.

Reply:
It's also kinda weird that the snackbar is never dismissed when the completes. I will change that, but please verify that it behaves correctly.